File: python/number-complement.py
class Solution(object):
    def findComplement(self, num):
        """
        :type num: int
        :rtype: int
        """
        if num < 0:
            return
        else:
            num = '{0:08b}'.format(num)
            print''.join(`1-int(x)`for x in nums)

File: ruby/number-complement.rb
# @param {Integer} num
# @return {Integer}

def find_complement(num)
	x = []
	for item in num.to_s(2).split('')		
		x.push((item == '0' ? '1' : '0'))
	end 	
	return x.join('').to_i(2)
end